boxes/boxes/__init__.py

```diff
@@ -0,0 +1,49 @@
+import collections
+import logging
+import pathlib
+from typing import List
+
+import boxes.utils.cmd
+
+
+class BoxRegister(type):
+    def __new__(meta, name, bases, attrs):
+        res_cls = super(BoxRegister, meta).__new__(meta, name, bases, attrs)
+        if not hasattr(meta, "_class_children"):
+            # A mapping from class to it's children
+            setattr(meta, "_class_children", collections.defaultdict(set))
+            setattr(meta, '_classes', {})
+
+        meta._classes[name] = res_cls
+        for base in bases:
+            meta._class_children[base.__name__].add(name)
+        return res_cls
+
+    def get_boxes():
+        boxes = {}
+        for box, box_cls in BoxRegister._classes.items():
+            if not len(BoxRegister._class_children[box]) and box != 'Box':
```

Comment:
IMO brzydka heura. Poza tym klasa pośrednia może zniknąć z wyniku jak zaimportujesz coś co po niej dziedziczy. Lepiej by było zamiast tej całej zabawy z metaklasami zrobić dekorator.

Reply:
Tak naprawdę, to taki dekorator też się wykona tylko na zaimportowanych plikach.

Reply:
I nie twierdzę, że metaclasy tutaj były sensownym rozwiązaniem ;p

Reply:
Tak, ale dekorator się nie od-wykona.
A tutaj możesz mieć klasę która jest zarejestrowana, ale jak zaimportujesz coś co po niej dziedziczy, to się od-rejestruje. IOW, z get_boxes mogą znikać rzeczy które już raz tam były, a to jest IMO niefajne.

Reply:
Wiem, to było pisane dla zabawy :). Zrobiłem zwykłego pistanego z palca dicta.